pkg/apis/servicecatalog/v1alpha1/types.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1 
+
+import (
+	kapi "k8s.io/client-go/1.5/pkg/api/v1"
+	kunversioned "k8s.io/client-go/1.5/pkg/api/unversioned"
+)
+
+// +nonNamespaced=true
+
+// Broker represents an entity that provides ServiceClasses for use in the
+// service catalog.
+type Broker struct {
+	kunversioned.TypeMeta
+	// Non-namespaced.  The name of this resource in etcd is in ObjectMeta.Name.
+	kapi.ObjectMeta
+
+	Spec   BrokerSpec
+	Status BrokerStatus
+}
+
+const (
+	// DescriptionKey is the key of an annotation that holds the brief
+	// description of an API resource
+	DescriptionKey = "alpha.service-catalog.kubernetes.io/description"
+)
+
+// BrokerSpec represents a description of a Broker.
+type BrokerSpec struct {
+	// The URL to communicate with the Broker via..
+	URL string
+
+	// Auth credentials should live in an api.Secret that
+	// is documented to have "username" and "password" keys
+	AuthUsername string
```

Comment:
@smarterclayton @liggitt who have seen APIs like this go wrong too and probably have a vague interest.